Makefile.am

```diff
@@ -5,20 +5,27 @@ AM_DISTCHECK_CONFIGURE_FLAGS = --enable-user-guides	\
 
 if PLATFORM_IS_LINUX_GENERIC
 PLATFORM_DIR = platform/linux-generic
+if WITH_TESTS
 PLATFORM_TEST_DIR = platform/linux-generic/test
 endif
+endif
 
 if WITH_EXAMPLES
 EXAMPLE_DIR = example
 endif
 
+if WITH_TESTS
+TEST_DIR = test
+HELPER_TEST_DIR = helper/test
+endif
+
 SUBDIRS = \
 	  include \
 	  $(PLATFORM_DIR) \
 	  helper \
-	  test \
+	  $(TEST_DIR) \
 	  $(PLATFORM_TEST_DIR) \
-	  helper/test \
+	  $(HELPER_TEST_DIR) \
```

Comment:
And `TEST_DIR`/`HELPER_TEST_DIR` are also not necessary. Just extend SUBDIRS under if condition

Reply:
+1

Reply:
Fixed in v2